From 5ec3a9150e845263d8341a9f4d910ce37fab0a37 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 18 Nov 2019 17:00:59 +0200
Subject: [PATCH] [SW-6739] hl: add platform init routine

It will be used for all what not done by BTL
In Goya2 there is no BTL so we need to put here all configurations used
to be done by BTL

Change-Id: Ib2a82427accdf9d0a47185d3d4f5ae6146428847
---

diff --git a/app/ppboot.c b/app/ppboot.c
index 6f6c4cb..60abcd5 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -89,6 +89,7 @@
 {
 	void (*preboot_start)(void) = (void *)PREBOOT_TEXT_BASE;
 
+	hl_plat_init();
 	hl_jtag_init();
 	preboot_sync_btl();
 	hl_uart_init();
diff --git a/gaudi/common.c b/gaudi/common.c
index f3652b5..48b82f8 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -191,3 +191,5 @@
 {
 	return 0;
 }
+
+void hl_plat_init(void) {}
diff --git a/goya/common.c b/goya/common.c
index 94e6723..adc5d7f 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -174,3 +174,5 @@
 {
 	return 0;
 }
+
+void hl_plat_init(void) {}
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 3549cc6..26153a8 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -145,3 +145,8 @@
 {
 	return 1;
 }
+
+void hl_plat_init(void)
+{
+	/* TBD: add relevant code we used to do in BTL */
+}
diff --git a/include/preboot.h b/include/preboot.h
index 1231e2f..1175ca0 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -81,6 +81,7 @@
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
 int hl_pcie_init_only(void);
+void hl_plat_init(void);
 
 void preboot_pcie_wait_for_link(void);
 void preboot_delay(void);
